intake_stac/catalog.py

```diff
@@ -97,21 +98,37 @@ def _load(self):
         """
         Load the STAC Catalog.
         """
-        for collection in self._stac_obj.collections():
-            self._entries[collection.id] = LocalCatalogEntry(
-                name=collection.id,
-                description=collection.title,
-                driver=StacCollection,
+        # NOTE: quick fix, logic/syntax likely could be improved!
+        lastCat = True
+
+        for subcatalog in self._stac_obj.children():
+            lastCat = False
+            self._entries[subcatalog.id] = LocalCatalogEntry(
+                name=subcatalog.id,
+                description=subcatalog.description,
+                driver=StacCatalog,
                 catalog=self,
-                args={'stac_obj': collection.filename},
+                args={'stac_obj': subcatalog.filename},
             )
 
+        # Ultimately we get to a Collection of Catalog of Items
+        if lastCat:
```

Comment:
can you help me understand the lastCat bit here? Will iterating over items high in the catalog hierarchy return items multiple levels down?

Reply:
that's correct. sat-utils.Catalog.items(), .catalogs(), and .collections() work recursively, but children() just gets the first level